src/core/namespace.c

```diff
@@ -1689,6 +1747,47 @@ int bind_mount_add(BindMount **b, size_t *n, const BindMount *item) {
         return 0;
 }
 
+MountEntry* mount_images_free_many(MountEntry *p, size_t *n) {
+        assert(n && (p || *n == 0));
+
+        for (size_t i = 0; i < *n; i++)
+                mount_entry_done(&p[i]);
+
+        *n = 0;
+        return mfree(p);
+}
+
+int mount_images_add(MountEntry **p, size_t *n, const MountEntry *item) {
+        _cleanup_free_ char *s = NULL, *d = NULL;
+        MountEntry *c;
+
+        assert(p);
+        assert(n);
+        assert(item);
+
+        s = strdup(mount_entry_source(item));
```

Comment:
Hmm, `mount_images_add` here shares a bunch of the same logic as `append_mount_images` above. Shouldn't there be a helper like `mount_image_copy_from(MountEntry** dst, MountEntry *src)` ?

Reply:
Sounds good, done